### PR TITLE
Update dart.yml

### PR DESCRIPTION
**Pull request by mit-mit**
Update Dart starter workflow to use `setup-dart` from the Dart team.

Note that I am a member of the Dart team and am more than happy to answer any questions about these changes.

Major advantages over using this new setup action are:

  1. Enables testing on more operating systems (Linux, Windows, and macOS)
  1. Offers more control over the Dart SDK version to test with (beta & dev channels, specific versions)

Thanks!

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

**Some general notes:**

- [ ] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [X] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [X] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [X] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

**Review by kevmoo — APPROVED** [review]

**Comment by kevmoo** [comment]
@mit-mit is a Product Manager at Google who is responsible for the Dart SDK.

**Review by devoncarew — APPROVED** [review]
Hey - a note for the actions-experience reviewers; I authored the previous version of this workflow; these changes lgtm :)

File: ci/dart.yml [hunk 1]
```diff
@@ -10,18 +15,13 @@ jobs:
...
-
     steps:
       - uses: actions/checkout@v2
 
-      - name: Print Dart SDK version
-        run: dart --version
+      # Note: This workflow uses the latest stable version of the Dart SDK.
+      # You can specify other versions if desired, see documentation here:
+      # https://github.com/dart-lang/setup-dart/blob/main/README.md
+      - uses: dart-lang/setup-dart@v1
```
Comment (andymckay) [inline ci/dart.yml:24]: Looks good, but I think we'll have to use the 40 char SHA of `setup-dart` for the moment.

Comment (mit-mit) [reply ci/dart.yml:24]: Using SHAs seems like a poor user experience compared to versions. Is there no other way around this?

Comment (mit-mit) [reply ci/dart.yml:24]: @andymckay, is SHAs a hard requirement?

Comment (joshmgross) [reply ci/dart.yml:24]: Outside of some code scanning workflows and actions in the https://github.com/actions Organization, this is a requirement. We're working on updating our Contributing guidelines to clarify this.

I agree this is not a great user experience, I'd recommend adding a comment above the `uses` for users to easily switch to a versioned tag. We do something similar in https://github.com/actions/starter-workflows/blob/main/ci/ruby.yml

```suggestion
      # - uses: dart-lang/setup-dart@v1
      - uses: dart-lang/setup-dart@9a04e6d73cca37bd455e0608d7e5092f881fd603
```

Comment (mit-mit) [reply ci/dart.yml]: OK, thanks. Took the commit suggestion.

**Review by joshmgross — APPROVED** [review]

**Pull request merged**